webapp/src/App.js

```diff
@@ -78,53 +84,63 @@ class DecksList extends React.Component {
 
 class DecksListDeck extends React.Component {
   proptypes = {
-    deck: PropTypes.object,
-    api: PropTypes.instanceOf(Api),
+    deck: PropTypes.object.isRequired,
+    deleteDeck: PropTypes.func.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = { isDeleted: false };
+  }
+
+  handleDeleteButtonClick() {
+    this.setState({ isDeleted: true });
```

Comment:
Setting the state on a component causes it to re-render itself automatically (that is a Reactjs thing)